Ignore positive integer errors from mbedtls_x509_crt_parse

borrrden opened the pull request:
Only in cases of importing system certs.  Some of them may be so old (or maybe so new?) as to be unable to be read by mbedTLS.  The behavior should not be to discard everything.  However when specifying our own chains we want everything to be valid.

borrrden committed:
Ignore positive integer errors from mbedtls_x509_crt_parse

Only in cases of importing system certs.  Some of them may be so old (or maybe so new?) as to be unable to be read by mbedTLS.  The behavior should not be to discard everything.  However when specifying our own chains we want everything to be valid.

snej commented on src/mbedtls_context.cpp:
```diff
@@ -441,19 +441,23 @@ namespace sockpp {
+        if (ret != 0) {
+	        if(ret < 0 || !partialOk) {
+		        log_mbed_ret(ret, "mbedtls_x509_crt_parse");
+				throw sys_error(ret);   //FIXME: Not an errno; use different exception?
```
In the `!partialOk` case, this is going to throw a `sys_error` whose code is the number of failed certs. That's pretty confusing because it'll be a small positive integer, indistinguishable from a POSIX error code. (As opposed to mbedTLS errors, which are negative.)
I suggest having it throw an mbedTLS 'invalid cert' error code.

borrrden committed:
Replace positive error when throwing

So that when throwing an exception up the chain, a positive integer doesn't get interpreted as an unrelated POSIX code.  Instead indicate that the verification failed.
